dask_sql/datacontainer.py

```diff
@@ -83,6 +83,26 @@ def rename(self, columns: Dict[str, str]) -> ColumnContainer:
 
         return cc
 
+    def rename_handle_duplicates(
+        self, from_columns: List[str], to_columns: List[str]
+    ) -> ColumnContainer:
+        """
+        Same as `rename` but additionally handles presence of
+        duplicates in `from_columns`
+        """
+        cc = self._copy()
+        for column_from, column_to in zip(from_columns, to_columns):
+            backend_column = self._frontend_backend_mapping[str(column_from)]
+            cc._frontend_backend_mapping[str(column_to)] = backend_column
+
+        columns = dict(zip(from_columns, to_columns))
```

Comment:
Would it make sense to create `columns` at the beginning since we need to iterate through `zip(from_columns, to_columns)` multiple times? Also feel like we could probably simplify the for loop to list comprehension since we can directly set `_frontend_backend_mapping`:

```suggestion
        columns = dict(zip(from_columns, to_columns))

        cc._frontend_backend_mapping.update({
            str(column_to): self._frontend_backend_mapping[str(column_from)]
            for column_from, column_to in columns.items()
        })
```

Reply:
I have updated the loop to be a list comprehension but moving the `columns` creation before updating `_frontend_backend_mapping` is the actual root cause of the issue. For ex:
```python
>>> from_cols = ['a', 'a']
>>> to_cols = ['b', 'd']
>>> dict(zip(from_cols, to_cols))
{'a': 'd'}
```

We don't want to eliminate `'b'` until we update `_frontend_backend_mapping`, that when `test_union_mixed` case starts failing.